Exiled.API/Features/Core/Generic/Singleton.cs

```diff
@@ -72,13 +81,7 @@ public static bool TryGet<TObject>(out TObject instance)
         /// <returns><see langword="true"/> if the instance was destroyed; otherwise, <see langword="false"/>.</returns>
         public static bool Destroy(T @object)
         {
-            if (Instances.TryGetValue(@object, out Singleton<T> _))
-            {
-                Instances[@object] = null;
-                return Instances.Remove(@object);
-            }
-
-            return false;
+            return Instances.Remove(@object);
```

Comment:
The old code, while may seem less efficient, is safer, considering the `ArgumentNullException` can occur creating issues further down the line. `TryGetValue` returns a boolean, which, in this context, stops bad, or null values from causing an exception.

Reply:
What do you mean by "effective"? Many optimizations can be made. Memory, speed, maintainability... ? 
Can you be a little more specific.

And no the old code is not safer for null
```cs
> Dictionary<Object, int> dic = new();
. var obj = new Object();
. dic[obj] = 10;
> dic.ContainsKey(obj)
true
> dic.ContainsKey(null)
System.ArgumentNullException: Value cannot be null. (Parameter 'key')
  + System.ThrowHelper.ThrowArgumentNullException(System.ExceptionArgument)
  + Dictionary<TKey, TValue>.FindValue(TKey)
  + Dictionary<TKey, TValue>.ContainsKey(TKey)
> dic.TryGetValue(null, out _)
. // do "_" and not "var _"
. // "var _" result to create a new var call "_"
System.ArgumentNullException: Value cannot be null. (Parameter 'key')
  + System.ThrowHelper.ThrowArgumentNullException(System.ExceptionArgument)
  + Dictionary<TKey, TValue>.FindValue(TKey)
  + Dictionary<TKey, TValue>.TryGetValue(TKey, out TValue)

```

if you whant to avoid `ArgumentNullException` do a check beffor.